### ISSUE TITLE
StreamMessageInput throws PlatformException(camera_access_denied)

### ISSUE THREAD

**Issue by Hwan-seok** [GetStream/stream-chat-flutter#1227]
### Which packages are you using?

stream_chat_flutter

### On what platforms did you experience the issue?

iOS, Android

### What version are you using?

stream_chat_flutter: 4.3.0

### What happened?

When trying to open camera or video directly at attachment in StreamMessageInput with denied permenatly permission, it throws PlatformException(camera_access_denied).

It should provide the callback when the user camera permission denied .

### Steps to reproduce

```bash
1. Create StreamMessageInput()
2. Tap attachment.
3. Tap camera attachment.
4. Tap deny button when permission-asking modal opens.
5. Tap camera attachment again.
```

### Relevant log output

```shell
[+1401 ms] [VERBOSE-2:ui_dart_state.cc(198)] Unhandled Exception: PlatformException(camera_access_denied, The user did not allow camera access., null, null)
[        ] #0      ImagePickerApi.pickImage
package:image_picker_ios/src/messages.g.dart:133
[        ] <asynchronous suspension>
[        ] #1      ImagePickerIOS.getImage
package:image_picker_ios/image_picker_ios.dart:168
[        ] <asynchronous suspension>
[        ] #2      StreamMessageInputState.pickFile
package:stream_chat_flutter/…/message_input/stream_message_input.dart:1585
[        ] <asynchronous suspension>
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by imtoori** [comment]
Thanks for reporting the issue
We need to explicitly ask for permissions
Adding this to the next sprint (starting today)

**Comment by imtoori** [comment]
- Ask explicitly for permission
- If the permission is not granted and the user taps the camera show a dialog redirecting to app settings

**Comment by CalHoll** [comment]
@imtoori on a similar note, when using `StreamMessageInput`, with `disableAttachments` set to true, when the widget builds it still requests to allow the app to access photos media and files on the device.

This should only happen when a user taps the attach button - and it shouldn't happen at all if the feature is disabled.

Is this thread ok for the permissions issue or should I open another issue?

**Comment by CalHoll** [comment]
> @CalHoll this is not happening for me, check that you're at least on version 4.3.0

My apologies, I've been locked on a previous version due to another dependency (4.1.0). Glad to hear it's resolved in newer releases, will upgrade ASAP. :-)

**Comment by imtoori** [comment]
closed by #1262
